guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md

```diff
@@ -0,0 +1,355 @@
+
+```
+import nodeResolve from 'rollup-plugin-node-resolve';
+import commonjs from 'rollup-plugin-commonjs';
+import typescript from 'rollup-plugin-typescript';
+import postCss from 'rollup-plugin-postcss';
+import externalGlobals from 'rollup-plugin-external-globals';
+
+export default [
+  {
+    input: 'src/index.tsx',
+    plugins: [
+      nodeResolve(),
+      commonjs(),
+      typescript(),
+      // Map imports from shared libraries (React, etc) to global variables exposed by Spinnaker.
+      externalGlobals(spinnakerSharedLibraries()),
+      // Import from .css, .less, and inject into the document <head></head>.
+      postCss(),
+    ],
+    output: [{ dir: 'dist', format: 'es', }]
+  }
+];
+
+function spinnakerSharedLibraries() {
+  const libraries = ['react', 'react-dom', '@spinnaker/core'];
+
+  function getGlobalVariable(libraryName) {
+    const prefix = 'spinnaker.plugins.sharedLibraries';
+    const sanitizedLibraryName = libraryName.replace(/[^a-zA-Z0-9_]/g, '_');
+    return `${prefix}.${sanitizedLibraryName}`;
+  }
+
+  return libraries.reduce((globalsMap, libraryName) => {
+    return { ...globalsMap, [ libraryName ]: getGlobalVariable(libraryName) }
+  }, {});
+}
+```
+
+`spinnakerSharedLibraries` pulls dependencies from Spinnaker. The libraries constant is a list of libraries that make the plugin work correctly.
+Items in this list must be from the [shared libraries](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/plugins/sharedLibraries.ts#L32) exposed to plugin creators.
+
+### Dependencies
+As mentioned above, Spinnaker exposes libraries for plugins to use. Define dependencies in package.json. For this example plugin, the dependencies are:
+
+```
+"@spinnaker/core": "0.0.432",
+"react": "^16.12.0",
+"react-dom": "^16.12.0"
+```
+
+## Writing The Frontend
+
+```
+import * as React from 'react';
+import { IStageTypeConfig, IStageConfigProps } from '@spinnaker/core';
+
+const customStage: IStageTypeConfig = {
+  label: 'Random Wait',
+  description: 'Stage that waits a random amount of time up to the max inputted',
+  key: 'randomWait',
+  component: RandomWaitStage,
+};
+
+function setMaxWaitTime(event: React.SyntheticEvent, props: IStageConfigProps) {
+  let target = event.target as HTMLInputElement;
+  props.updateStageField({'maxWaitTime': target.value});
+}
+
+// Our stage component
+function RandomWaitStage(props: IStageConfigProps) {
+  return (
+    <div>
+      <label>
+          Max Time To Wait
+          <input value={props.stage.maxWaitTime} onChange={(e) => setMaxWaitTime(e, props)} id="maxWaitTime" />
+      </label>
+    </div>
+  );
+}
+
+const plugin = {
+  name: 'randomWait',
+  stages: [customStage],
+};
+
+export { plugin };
+```
+
+### IStageTypeConfig
+Define Spinnaker Stages with IStageTypeConfig. Required [options:](https://github.com/spinnaker/deck/blob/abac63ce5c88b809fcf5ed1509136fe96489a051/app/scripts/modules/core/src/domain/IStageTypeConfig.ts)
```

Comment:
```suggestion
Define Spinnaker Stages with IStageTypeConfig. Required [options](https://github.com/spinnaker/deck/blob/abac63ce5c88b809fcf5ed1509136fe96489a051/app/scripts/modules/core/src/domain/IStageTypeConfig.ts):
```

Reply:
removed this and added as code comments  in the pf4jStagePlugin repo